appservice/src/deploy/deployToStorageAccount.ts

```diff
@@ -39,10 +40,28 @@ async function createBlobServiceClient(client: SiteClient): Promise<BlobServiceC
     // Use same storage account as AzureWebJobsStorage for deployments
     const azureWebJobsStorageKey: string = 'AzureWebJobsStorage';
     const settings: WebSiteManagementModels.StringDictionary = await client.listApplicationSettings();
-    if (settings.properties && settings.properties[azureWebJobsStorageKey]) {
-        return BlobServiceClient.fromConnectionString(settings.properties[azureWebJobsStorageKey]);
+    let connectionString: string | undefined = settings.properties && settings.properties[azureWebJobsStorageKey];
+    if (connectionString) {
+        try {
+            return BlobServiceClient.fromConnectionString(connectionString);
```

Comment:
Since it's required now, shouldn't we just always add the endpointSuffix if it doesn't have one?  If we know creating the BlobServiceClient will fail without it, seems like we're just wasting time by trying first.

Reply:
I consider what I'm doing to be a bit of "black box magic". When I do that, I prefer to always leave a path so that the user can avoid the magic if it doesn't work for them. In this case I can just avoid the magic by default since "fromConnectionString" does all the validation up front before trying to connect to Azure (aka it's super fast). Basically, I want to protect against the scenario where my code isn't perfect and doesn't handle all edge cases correctly haha.